docs/user-guide/reminders-and-external-events.rst

```diff
@@ -0,0 +1,257 @@
+Thus, we define an intent ``ask_remind_call`` with some NLU data,
+
+.. code-block:: md
+
+  ## intent:ask_remind_call
+  - remind me to call [Albert](name)
+  - remind me to call [Susan](name)
+  - later I have to call [Daksh](name)
+  - later I have to call [Anna](name)
+  ...
+
+and connect this intent with a new custom action ``action_set_reminder``.
+We could make this connection by providing training stories (recommended for more complex assistants), or using the :ref:`mapping-policy`.
+
+The custom action ``action_set_reminder`` should schedule a reminder that, 5 seconds later, triggers an intent ``EXTERNAL_reminder`` with all the entities that the user provided in his/her last message (similar to an external event):
+
+.. code-block:: python
+
+  class ActionSetReminder(Action):
+    """Schedules a reminder, supplied with the last message's entities."""
+
+    def name(self) -> Text:
+        return "action_set_reminder"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        dispatcher.utter_message("I will remind you in 5 seconds.")
+
+        date = datetime.datetime.now() + datetime.timedelta(seconds=5)
+        entities = tracker.latest_message.get("entities")
+
+        reminder = ReminderScheduled(
+            "EXTERNAL_reminder",
+            trigger_date_time=date,
+            entities=entities,
+            name="my_reminder",
+            kill_on_user_message=False,
+        )
+
+        return [reminder]
+
+
+Note, that this requires the ``datetime`` and ``rasa-sdk.events`` packages.
+For details, have a look at the ``reminderbot`` example under ``rasa/examples/reminderbot``.
+
+Finally, we define another custom action ``action_react_to_reminder`` and link it to the ``EXTERNAL_reminder`` intent:
+
+.. code-block:: md
+
+  - EXTERNAL_reminder:
+    triggers: action_react_to_reminder
+
+where the ``action_react_to_reminder`` is
+
+.. code-block:: python
+
+  class ActionReactToReminder(Action):
+    """Reminds the user to call someone."""
+
+    def name(self) -> Text:
+        return "action_react_to_reminder"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        name = next(tracker.get_latest_entity_values("name"), None) or "someone"
+        dispatcher.utter_message(f"Remember to call {name}!")
+
+        return []
+
+Instead of a custom action, we could also have used a simple response template.
+But here we want to make use of the fact that the reminder can carry entities, and we can process the entities in this custom action.
+
+.. warning::
+
+  Reminders are cancelled whenever you shutdown rasa.
+
+
+.. warning::
+
+  Reminders currently (Rasa 1.7) don't work in `rasa shell`.
+  Use `rasa x` instead.
```

Comment:
Somebody could fix the bug ;)

Reply:
@erohmensing I have the same problem again when writing the README. What should I do? @tmbo How hard is it to fix the reminders-dont-work-in-shell bug?

Reply:
mhm now that we have SocketIO we could switch to that instead of the REST channel. This would allow us to push messages back and forth. in any case it is a bit of work and nothing we can quickly fix

Reply:
Ok. Then I refer people to the rasa server installation instead, for now.

Reply:
@tmbo I thought socketio is the only channel other than REST that doesn't support proactive intents?